src/query/functions/temporal/aggregation_test.go

```diff
@@ -131,73 +131,183 @@ var testCases = []testCase{
+		},
+		afterAllBlocks: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+	},
+	{
+		name:   "count_over_time",
+		opType: CountTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 0},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), 0},
+		},
+		afterAllBlocks: [][]float64{
+			{0, 0, 0, 0, 0},
+			{0, 0, 0, 0, 0},
+		},
+	},
+	{
+		name:   "min_over_time",
+		opType: MinTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+		afterAllBlocks: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+	},
+	{
+		name:   "max_over_time",
+		opType: MaxTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+		afterAllBlocks: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+	},
+	{
+		name:   "sum_over_time",
+		opType: SumTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+		afterAllBlocks: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+	},
+	{
+		name:   "stddev_over_time",
+		opType: StdDevTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+		afterAllBlocks: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+	},
+	{
+		name:   "stdvar_over_time",
+		opType: StdVarTemporalType,
+		afterBlockOne: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+		afterAllBlocks: [][]float64{
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+			{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		},
+	},
+}
+
+func TestAggregationAllNaNs(t *testing.T) {
+	v := [][]float64{
+		{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+		{math.NaN(), math.NaN(), math.NaN(), math.NaN(), math.NaN()},
+	}
+	testAggregation(t, testCasesNaNs, v)
+}
+
+func testAggregation(t *testing.T, testCases []testCase, vals [][]float64) {
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			var (
+				values [][]float64
+				bounds block.Bounds
+			)
+
+			if vals == nil {
```

Comment:
Can you just pass in values instead of doing this generation thing? It's really bad for comprehension and understanding why the results are what they are